src/main/java/org/opensrp/web/rest/v2/SettingResource.java

```diff
@@ -110,25 +114,40 @@ public void setSettingService(SettingService settingService, PhysicalLocationSer
 		String teamId = RestUtils.getStringFilter(AllConstants.Event.TEAM_ID, request);
 		String identifier = RestUtils.getStringFilter(SETTING_IDENTIFIER, request);
 		boolean resolveSettings = RestUtils.getBooleanFilter(AllConstants.Event.RESOLVE_SETTINGS, request);
+		String metadataVersion = RestUtils.getStringFilter(METADATA_VERSION, request);
+		String limit = RestUtils.getStringFilter(LIMIT, request);
 		Map<String, TreeNode<String, Location>> treeNodeHashMap = null;
 
 		if (StringUtils.isBlank(team) && StringUtils.isBlank(providerId) && StringUtils.isBlank(locationId)
-				&& StringUtils.isBlank(teamId) && StringUtils.isBlank(team) && StringUtils.isBlank(serverVersion)) {
+				&& StringUtils.isBlank(teamId) && StringUtils.isBlank(team) && StringUtils.isBlank(serverVersion)
+				&& StringUtils.isBlank(metadataVersion)) {
```

Comment:
Wont this break the current functionality `StringUtils.isBlank(metadataVersion)`?

Reply:
This check is to ensure that all parameters can not be null. Hence, added meta version field here as well. @bennsimon 
Please let me know if there is anything incorrect. 
![settingsv2get](https://user-images.githubusercontent.com/62061579/120990292-d573ec80-c799-11eb-8bfb-f756894f1d24.png)

Reply:
This is a new parameter so for the devs that were supplying the other params only it should work.

Reply:
Yes, `metadata_version` and `limit` request params are not required fields. This check is to ensure that all request parameters can not be null, we need to supply at least a single request param from the given list. Thus added the check. 
Should I remove validation from here? @bennsimon

Reply:
leave it